web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -14,24 +14,83 @@
  * limitations under the License.
  */
 
-import { Component, Inject, OnInit } from '@angular/core';
+import { Component, Inject, OnDestroy } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ProjectService } from '../../services/project/project.service';
 import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Project } from '../../shared/models/project.model';
+import { Layer } from '../../shared/models/layer.model';
+import { Subscription } from 'rxjs';
+import { DataStoreService } from '../../services/data-store/data-store.service';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-layer-dialog',
   templateUrl: './layer-dialog.component.html',
   styleUrls: ['./layer-dialog.component.css'],
 })
-export class LayerDialogComponent {
+export class LayerDialogComponent implements OnDestroy {
   layerId: string;
-
+  layer?: Layer;
+  projectId?: string;
+  activeProject$: Observable<Project>;
+  subscription: Subscription = new Subscription();
   constructor(
     // tslint:disable-next-line:no-any
     @Inject(MAT_DIALOG_DATA) data: any,
-    private dialogRef: MatDialogRef<LayerDialogComponent>
+    private dialogRef: MatDialogRef<LayerDialogComponent>,
+    private projectService: ProjectService,
+    private dataStoreService: DataStoreService,
+    private router: Router
   ) {
-    this.layerId = data.layerId!;
     // Disable closing on clicks outside of dialog.
     dialogRef.disableClose = true;
+    this.layerId = data.layerId;
+    this.activeProject$ = this.projectService.getActiveProject$();
+    this.subscription.add(
+      this.activeProject$.subscribe(project => {
+        this.onProjectLoaded(project);
+      })
+    );
+  }
+
+  onProjectLoaded(project: Project) {
+    if (this.layerId === ':new') {
+      this.layerId = this.dataStoreService.generateId();
+      this.layer = {
+        id: this.layerId,
+      };
+    } else {
+      this.layer = project.layers.get(this.layerId);
+    }
+    if (!this.layer) {
+      throw Error('No layer exists');
+    }
+    this.projectId = project.id;
+  }
+
+  onSave() {
+    if (!this.projectId) {
+      throw Error('No Project exists with that id');
```

Comment:
This will only happen if we somehow end up here before the project is loaded (not when no project exists with this id). This could happen if the user opens a bookmark with the layer dialog open and clicks "Save" before the project config finished loading. Ideally, we wouldn't open the layer dialog until the project dialog is loaded, which can be achieved by waiting for a "project' to be loaded before calling `showEditLayerDialog()` in `MainPageComponent`.

For now, could you kindly update the error message to 'Project not yet loaded' and add a TODO to wait for project to load before showing dialog?